hivemq-edge/src/frontend/src/extensions/datahub/components/helpers/FunctionCreatableSelect.tsx

```diff
@@ -0,0 +1,111 @@
+import { FC, useCallback } from 'react'
+import { OptionProps, SingleValueProps, chakraComponents, ActionMeta, OnChangeValue, Select } from 'chakra-react-select'
+import { getChakra } from '@rjsf/chakra-ui/lib/utils'
+import { labelValue, WidgetProps } from '@rjsf/utils'
+import { HStack, VStack, Text, FormLabel, FormControl } from '@chakra-ui/react'
+import { useTranslation } from 'react-i18next'
+
+import { FunctionSpecs } from '../../types.ts'
+import useDataHubDraftStore from '../../hooks/useDataHubDraftStore.ts'
+
+const SingleValue = (props: SingleValueProps<FunctionSpecs>) => {
+  return (
+    <chakraComponents.SingleValue {...props}>
+      <Text>{props.data.functionId}</Text>
+    </chakraComponents.SingleValue>
+  )
+}
+
+const Option = (props: OptionProps<FunctionSpecs>) => {
+  const { t } = useTranslation('datahub')
+  const { isSelected, ...rest } = props
+  const [dd] = props.getValue()
+
+  // @ts-ignore
+  const { __isNew__ } = props.data
+  if (__isNew__) {
+    return <chakraComponents.Option {...props}>{props.children}</chakraComponents.Option>
+  }
+
+  const { isTerminal, isDataOnly } = props.data.metadata || {}
+
+  return (
+    <chakraComponents.Option {...rest} isSelected={dd && dd.functionId === props.data.functionId}>
+      <VStack w="100%" alignItems="stretch" gap={0}>
+        <HStack>
+          <Text as="b" flex={1}>
+            {props.data.functionId}
+          </Text>
+          <HStack>
+            {isTerminal && <Text fontSize="sm">{t('workspace.function.isTerminal')}</Text>}
+            {isDataOnly && <Text fontSize="sm">{t('workspace.function.isDataOnly')}</Text>}
+          </HStack>
+        </HStack>
+        <Text fontSize="sm">{props.data.schema?.description}</Text>
+      </VStack>
+    </chakraComponents.Option>
+  )
+}
+
+const getValue = (props: WidgetProps) => {
+  const options = props.options.enumOptions
+  const defs = props.registry.rootSchema.definitions
+
+  if (!defs || !options) return undefined
+
+  const deg = defs[props.value]
+  if (!deg) return undefined
+
+  return {
+    schema: deg,
+    functionId: props.value,
+    // @ts-ignore
+    ...deg.metadata,
+  }
+}
```

Comment:
suggestion: let's give them full names, I tied to get context out of this but I've no clue what they mean

```suggestion
  const rootSchemaDefitions = props.registry.rootSchema.definitions

  if (!rootSchemaDefitions || !options) return undefined

  const rootSchemaDefitionItem = rootSchemaDefitions[props.value]
  if (!rootSchemaDefitionItem) return undefined

  return {
    schema: rootSchemaDefitionItem,
    functionId: props.value,
    // @ts-ignore
    ...rootSchemaDefitionItem.metadata,
  }
}
```

Reply:
Agree. I got no excuse these days, not with the IDE doing a good job of proposing meaningful variable names

![Screenshot 2024-02-01 at 12 51 24](https://github.com/hivemq/hivemq-edge/assets/2743481/228b63ca-adbb-43a0-82d1-63439555ffaf)